sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/InMemoryTableWithV2FilterCatalog.scala

```diff
@@ -43,4 +43,8 @@ class InMemoryTableWithV2FilterCatalog extends InMemoryTableCatalog {
     namespaces.putIfAbsent(ident.namespace.toList, Map())
     table
   }
+
+  override def createTable(ident: Identifier, tableInfo: TableInfo): Table = {
+    createTable(ident, tableInfo.columns(), tableInfo.partitions(), tableInfo.properties)
```

Comment:
Since the parameter is `TableInfo`, don't we need to pass `constraints` here. Cannot this `TableInfo` contain `constraints`?

Reply:
By default, `createTable(ident: Identifier, tableInfo: TableInfo)` falls back to the derecated create table methods
However, the base class `InMemoryTableCatalog` overrides `createTable(ident: Identifier, tableInfo: TableInfo)`, and now we need to modify all the subclasses.
Otherwise, create table execution will use the base class's `createTable(ident: Identifier, tableInfo: TableInfo)`

Reply:
We should have a follow on unifying the create table overrides in tests:
<img width="918" alt="image" src="https://github.com/user-attachments/assets/d2af0efc-cf24-4695-84ac-0fe500508b8d" />